backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/sort/nested/onthefly/doubleval/impl/SortedNumericDoubleValues.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.types.sort.nested.onthefly.doubleval.impl;
+
+import java.io.IOException;
+
+import org.apache.lucene.index.NumericDocValues;
+import org.apache.lucene.index.SortedNumericDocValues;
+
+/**
+ * Clone of {@link SortedNumericDocValues} for double values.
+ * <p>
+ * Copied and adapted from {@code org.elasticsearch.index.fielddata.SortedNumericDoubleValues} class
+ * of <a href="https://github.com/elastic/elasticsearch">Elasticsearch project</a>.
+ */
+public abstract class SortedNumericDoubleValues {
```

Comment:
I think this, and all the classes it relies on, is not currently necessary, as we could work directly on longs all along.
Diffentiating longs and doubles only makes sense when handling sorts on multi-valued fields and sorting on the average (because I don't think we can compute an avergae directly on the long-encoded representation of doubles).

Anyway... let's keep it that way if you want, I guess it will be useful later.